Tests/XCUITests/JumpBackInTests.swift

```diff
@@ -0,0 +1,175 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import XCTest
+
+class JumpBackInTests: BaseTestCase {
+    func closeKeyboard() {
+        waitForExistence(app.buttons["urlBar-cancel"], timeout: TIMEOUT)
+        navigator.performAction(Action.CloseURLBarOpen)
+        navigator.nowAt(NewTabScreen)
+    }
+
+    override func setUp() {
+        super.setUp()
+
+        closeKeyboard()
+
+        // Since I can't scroll on M1, closing the "Switch Your Default Browser"
+        // box makes the "Jump Back In" section visible without scrolling in the
+        // last part of the test.
+        if app.staticTexts["Switch Your Default Browser"].exists {
+            app.buttons["Close"].firstMatch.tap()
+        }
+
+        // "Jump Back In" is enabled by default. See Settings -> Homepage
+        navigator.goto(HomeSettings)
+        XCTAssertEqual(app.switches["Jump Back In"].value as! String, "1")
+
+        navigator.goto(NewTabScreen)
+    }
+
+    func testJumpBackInSection() {
+        // Open a tab and visit a page
+        navigator.openURL("https://www.example.com")
+        waitUntilPageLoad()
+
+        // Open a new tab
+        navigator.goto(TabTray)
+        navigator.performAction(Action.OpenNewTabFromTabTray)
+        closeKeyboard()
+
+        // "Jump Back In" section is displayed
+        waitForExistence(app.staticTexts["Jump Back In"])
+
+        // The contextual hint box is not displayed consistently, so
+        // I don't test for its existence.
+    }
+
+    func testGroupedTabs() {
+        // Enable "Tab Groups" from Settings -> Tabs
+        navigator.goto(TabsSettings)
+        navigator.performAction(Action.ToggleTabGroups)
+        navigator.goto(SettingsScreen)
+        app.navigationBars.buttons["Done"].tap()
+        navigator.nowAt(NewTabScreen)
+
+        // Create 3 groups in tab tray
+        let groups = ["test1", "test2", "test3"]
+        for group in groups {
+            for _ in 1...3 {
```

Comment:
The steps on TestRail says repeat twice, but I found that repeating 3 times can ensure "Test3" shows up as a group under "Jump Back In" section.

Reply:
While debugging this test, I think it is too long.. even though the manual test says to create 3 groups, just with 1 created we can  check the functionality...